src/main/java/org/jetbrains/kotlinx/lincheck/strategy/TrustedAtomicPrimitives.kt

```diff
@@ -0,0 +1,61 @@
+/*-
+ * #%L
+ * Lincheck
+ * %%
+ * Copyright (C) 2019 - 2020 JetBrains s.r.o.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+package org.jetbrains.kotlinx.lincheck.strategy
+
+/**
+ * Some atomic primitives are common and can be analyzed from a higher level of abstraction
+ * or can not be transformed (i.e, Unsafe or AFU).
+ * We don't transform them and improve the output of lincheck for them.
+ * For example, in the execution instead of a code location in AtomicLong.get() method
+ * we could just print the code location where the method is called.
+ */
+internal object TrustedAtomicPrimitives {
```

Comment:
Cool feature!

Reply:
Let's expand this feature on other trusted blocks. E.g., you use `ConcurrentHashMap` in your implementation and you believe that its operations are atomic (e.g., `put`, `get`, `remove` but not `size` are linearizable). Thus, you can exclude the `ConcurrentHashMap` from the search and save resources for more detailed analysis. 

To be short, we need contracts here.